sources/packages/backend/apps/api/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -103,4 +103,49 @@ export class CASSupplierService {
 
     return student.casSupplier;
   }
+
+  /**
+   * Retries CAS supplier info for a student.
+   * Inserts a new CAS Supplier record for the student with PendingSupplierVerification status.
+   * @param studentId student id.
+   * @param auditUserId user id for the record creation.
+   */
+  async retryCASSupplier(
+    studentId: number,
+    auditUserId: number,
+  ): Promise<CASSupplier> {
+    const student = await this.studentService.getStudentById(studentId);
+    const auditUser = { id: auditUserId } as User;
+    if (!student) {
+      throw new CustomNamedError("Student not found.", STUDENT_NOT_FOUND);
+    }
+    const studentCasSupplier: CASSupplier = await this.casSupplierRepo.findOne({
+      select: {
+        id: true,
+        createdAt: true,
+        supplierStatus: true,
+      },
+      where: {
+        student: { id: studentId },
+      },
+      order: {
+        createdAt: "DESC",
+      },
+    });
+    if (
+      studentCasSupplier?.supplierStatus ===
+      SupplierStatus.PendingSupplierVerification
+    ) {
+      throw new UnprocessableEntityException(
+        `There is already a CAS Supplier for this student in ${SupplierStatus.PendingSupplierVerification} status.`,
+      );
+    }
+    const casSupplier = new CASSupplier();
+    casSupplier.supplierStatus = SupplierStatus.PendingSupplierVerification;
+    casSupplier.supplierStatusUpdatedOn = new Date();
+    casSupplier.isValid = false;
+    casSupplier.creator = auditUser;
+    casSupplier.student = student;
```

Comment:
We can have a centralized method to enforce the creation of the pending supplier.
This code is the same from the student creation below and they should probably be evolving together if needed.
https://github.com/bcgov/SIMS/blob/ba003cf3806d691d1dd3ce68ab0d738f7ab1a54f/sources/packages/backend/apps/api/src/services/student/student.service.ts#L227-L232